test/units/matching/HyperdriveMatchingEngineTest.t.sol

```diff
@@ -0,0 +1,1535 @@
+            );
+            vm.revertTo(snapshotId);
+        }
+
+        // Ensure that order matching fails when the long order was cancelled.
+        {
+            uint256 snapshotId = vm.snapshot();
+            vm.stopPrank();
+            vm.startPrank(bob);
+            IHyperdriveMatchingEngine.OrderIntent[]
+                memory orders = new IHyperdriveMatchingEngine.OrderIntent[](1);
+            orders[0] = shortOrder;
+            matchingEngine.cancelOrders(orders);
+            vm.expectRevert(
+                IHyperdriveMatchingEngine.AlreadyCancelled.selector
+            );
+            matchingEngine.matchOrders(
+                // long order
+                longOrder,
+                // short order
+                shortOrder,
+                // LP amount
+                2_000_000e18,
+                // add liquidity options
+                IHyperdrive.Options({
+                    asBase: true,
+                    destination: address(matchingEngine),
+                    extraData: ""
+                }),
+                // remove liquidity options
+                IHyperdrive.Options({
+                    asBase: true,
+                    destination: address(matchingEngine),
+                    extraData: ""
+                }),
+                // fee recipient
+                celine,
+                true
+            );
+            vm.revertTo(snapshotId);
+        }
+
+        // Ensure that order matching fails when both orders were cancelled.
+        {
+            uint256 snapshotId = vm.snapshot();
+            vm.stopPrank();
+            vm.startPrank(alice);
+            IHyperdriveMatchingEngine.OrderIntent[]
+                memory orders = new IHyperdriveMatchingEngine.OrderIntent[](1);
+            orders[0] = longOrder;
+            matchingEngine.cancelOrders(orders);
+            vm.startPrank(bob);
+            orders = new IHyperdriveMatchingEngine.OrderIntent[](1);
+            orders[0] = shortOrder;
+            matchingEngine.cancelOrders(orders);
+            vm.expectRevert(
+                IHyperdriveMatchingEngine.AlreadyCancelled.selector
+            );
+            matchingEngine.matchOrders(
+                // long order
+                longOrder,
+                // short order
+                shortOrder,
+                // LP amount
+                2_000_000e18,
+                // add liquidity options
+                IHyperdrive.Options({
+                    asBase: true,
+                    destination: address(matchingEngine),
+                    extraData: ""
+                }),
+                // remove liquidity options
+                IHyperdrive.Options({
+                    asBase: true,
+                    destination: address(matchingEngine),
+                    extraData: ""
+                }),
+                // fee recipient
+                celine,
+                true
+            );
+            vm.revertTo(snapshotId);
```

Comment:
```suggestion
```

Reply:
I add these to avoid issues if we add more cases. I'm going to leave this.